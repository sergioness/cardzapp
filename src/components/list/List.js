import React from 'react';
import Card from '../card/Card';

class List extends React.PureComponent {
    render() {
        const values = this.props.values;
        return (
            <React.Fragment>
            {
                values.map((value) => {
                    return (<Card key={value.id} info={value}/>);
                })
            }
            </React.Fragment>
        );
    }
};
export default List;