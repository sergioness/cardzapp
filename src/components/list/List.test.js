import Enzyme from 'enzyme';
import React from 'react';
import List from './List';
import toJson from 'enzyme-to-json';

describe('Component \'List\'', () => {
  it('shallow matches the snapshot', () => {
    const mockValues = [
      {
        id: 0,
        name: 'dummy card',
        email: 'card@card.com'
      }
    ];
    const shallowList = Enzyme.shallow(<List values={mockValues}/>);
    expect(toJson(shallowList, {mode: 'deep'})).toMatchSnapshot();
  });
});
