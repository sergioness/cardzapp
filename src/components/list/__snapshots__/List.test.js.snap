// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component 'List' shallow matches the snapshot 1`] = `
<Fragment>
  <Card
    info={
      Object {
        "email": "card@card.com",
        "id": 0,
        "name": "dummy card",
      }
    }
    key="0"
  />
</Fragment>
`;
